projects/adc_periodic_reader/inc/adc_periodic_reader.h

```diff
@@ -0,0 +1,47 @@
+#pragma once
+

```

Comment:
List out all the dependencies for this module. e.g. requires soft_timers to be initialized.